semgrep/semgrep/evaluation.py

```diff
@@ -12,73 +11,35 @@
 logger = getLogger(__name__)
 
 
-def compare_propagated_metavariable(
-    _range: Range,
-    pattern_match: PatternMatch,
-    metavariable: str,
-) -> bool:
-
-    return (
-        metavariable in _range.propagated_metavariables
-        and metavariable in pattern_match.metavariable_uids
-        and _range.propagated_metavariables[metavariable]
-        == pattern_match.metavariable_uids[metavariable]
-    )
-
-
 def create_output(
     rule: Rule,
     pattern_matches: List[PatternMatch],
     valid_ranges_to_output: Optional[Set[Range]] = None,
 ) -> List[RuleMatch]:
     output = []
-
-    if valid_ranges_to_output is None:
```

Comment:
This is always None since the only place we call create_output does not set this value. So this and line 52 are essentially no-ops now.